### ISSUE TITLE
Missing async close on io::Write

### ISSUE THREAD

**Issue by jocutajar** [async-rs/async-std#977]
Hello,

I would like to close IO asynchronously.

Please add close to WriteExt as seen in futures: https://docs.rs/futures-lite/1.12.0/futures_lite/io/trait.AsyncWriteExt.html

**Comment by Fishrock123** [comment]
Are you asking for `AsyncDrop`?

**Comment by jbr** [comment]
I think it's just that futures_lite exposes a close function, but async-std's WriteExt [does not](https://docs.rs/async-std/1.9.0/async_std/io/prelude/trait.WriteExt.html) although async-std's Write does include a [poll_close](https://docs.rs/async-std/1.9.0/async_std/io/trait.Write.html#tymethod.poll_close). futures_lite::AsyncWriteExt _should_ be compatible with async-std's Write, which I believe is a reexport of [`futures::AsyncWrite`](https://docs.rs/futures-io/0.3.15/futures_io/trait.AsyncWrite.html), so I think it's possible to use the AsyncWriteExt from futures-lite with async-std's Write, but I believe this issue is specifically about adding a close method to the async-std one in order to make poll_close useful

**Comment by jocutajar** [comment]
@Fishrock123, no. And I imagine that would be pretty tough to do, wouldn't it?

@jbr, exactly. I could use futures, but why pull yet another dependency? I want to `io.close().await?;`
